common/cpp/react/renderer/components/rnscreens/RNSScreenStackHeaderSubviewComponentDescriptor.h

```diff
@@ -35,9 +35,48 @@ class RNSScreenStackHeaderSubviewComponentDescriptor final
         shadowNode.getState());
     auto stateData = state->getData();
 
+#if defined(ANDROID)
     if (!isSizeEmpty(stateData.frameSize)) {
       layoutableShadowNode.setSize(stateData.frameSize);
     }
+#else
+    auto mostRecentState = std::static_pointer_cast<
+        const RNSScreenStackHeaderSubviewShadowNode::ConcreteState>(
+        shadowNode.getMostRecentState());
+
+    auto mostRecentStateData = mostRecentState->getData();
+
+    std::printf(
+        "SubviewCD [%d] adopt frameSize {%.2lf, %.2lf}, mostRecent {%.2lf, %.2lf}\n",
+        shadowNode.getTag(),
+        stateData.frameSize.width,
+        stateData.frameSize.height,
+        mostRecentStateData.frameSize.width,
+        mostRecentStateData.frameSize.height);
+
+    if (!isSizeEmpty(stateData.frameSize)) {
+      // This is a ugly hack to workaround issue with dynamic content change.
+      // When the size of this shadow node contents (children) change due to JS
+      // update, e.g. new icon is added, if the size is set for the yogaNode
+      // corresponding to this shadow node, the enforced size will be used
+      // and the size won't be updated by Yoga to reflect the contents size
+      // change ->
+      // -> host tree won't get layout metrics update -> we won't trigger native
+      // layout -> the views in header will be positioned incorrectly. Here we
+      // try to detect, whether this shadow node is cloned as a result of state
+      // update (we assume that only source if the state updates is HostTree) or
+      // other change.
+      if (stateData.frameSize != mostRecentStateData.frameSize) {
+        std::printf("SubviewCD [%d] adopt APPLY\n", shadowNode.getTag());
+        layoutableShadowNode.setSize(stateData.frameSize);
+      } else {
+        // If the state has not changed we assign undefined size, to allow Yoga
+        // to recompute the shadow node layout.
+        std::printf("SubviewCD [%d] adopt ZERO\n", shadowNode.getTag());
+        layoutableShadowNode.setSize({YGUndefined, YGUndefined});
+      }
```

Comment:
This causes regression in header truncation (`TestHeaderTitle`).

Imagine case with two header subviews present in the navigation bar. Now, when one updates, the other receives `YGUndefined` values as its size (as it's state does not change), effectively ruining header subview sizing.